packages/rum/src/apm-base.js

```diff
@@ -101,6 +101,11 @@ class ApmBase {
     return !this._disable
   }
 
+  observe(name, fn) {
+    const configService = this.serviceFactory.getService('ConfigService')
+    configService.events.observe(name, fn)
```

Comment:
I think the event handler should be placed in a common place, since we can use it to listen to and send any event, whether related to transactions or not (e.g for span events or errors as you mentioned). 

Taking a step back, I think we need a common place to put these top level functions, whether it should be config service or not I don't have a strong opinion on. it might also be a naming issue, i.e. if we refactor `ConfigService` to be `AgentService`, that too is a fine approach IMO.

Reply:
Decided to introduce `AgentService` in a separate PR.